src/data/list/chain.lean

```diff
@@ -105,21 +105,31 @@ begin
   { simp [H _ _ _ _ (rel_of_chain_cons hl₂), l_ih _ _ (chain_of_chain_cons hl₂)] }
 end
 
-theorem chain_of_pairwise {a : α} {l : list α} (p : pairwise R (a :: l)) : chain R a l :=
+protected lemma pairwise.chain (p : pairwise R (a :: l)) : chain R a l :=
 begin
   cases pairwise_cons.1 p with r p', clear p,
   induction p' with b l r' p IH generalizing a, {exact chain.nil},
   simp only [chain_cons, forall_mem_cons] at r,
   exact chain_cons.2 ⟨r.1, IH r'⟩
 end
 
+protected lemma chain.pairwise (tr : transitive R) :
+  ∀ {a : α} {l : list α}, chain R a l → pairwise R (a :: l)
+| a [] chain.nil := pairwise_singleton _ _
+| a _ (@chain.cons _ _ _ b l h hb) := hb.pairwise.cons begin
+    simp only [mem_cons_iff, forall_eq_or_imp, h, true_and],
+    exact λ c hc, tr h (rel_of_pairwise_cons hb.pairwise hc),
+  end
+
 theorem chain_iff_pairwise (tr : transitive R) {a : α} {l : list α} :
   chain R a l ↔ pairwise R (a :: l) :=
-⟨λ c, begin
-  induction c with b b c l r p IH, {exact pairwise_singleton _ _},
-  apply IH.cons _, simp only [mem_cons_iff, forall_eq_or_imp, r, true_and],
-  show ∀ x ∈ l, R b x, from λ x m, (tr r (rel_of_pairwise_cons IH m)),
-end, chain_of_pairwise⟩
+⟨chain.pairwise tr, pairwise.chain⟩
+
+protected lemma chain.sublist [is_trans α R] (hl : l₂.chain R a) (h : l₁ <+ l₂) : l₁.chain R a :=
```

Comment:
Do we have any convention about `(h : transitive R)` vs `[is_trans α R]`?

Reply:
Not that I know of, unfortunately...

Reply:
My understanding was the `is_trans` was introduced as part of the "The big algebra typeclass refactor" that never happened.